lib/tasks/demo.rake

```diff
@@ -14,84 +14,129 @@ namespace :demo do
-
-      count.times do
-        FactoryGirl.create(:package, :with_item, :package_with_locations, :with_inventory_number)
-      end
-      count.times do
-        FactoryGirl.create(:package, :stockit_package, :with_inventory_number , :package_with_locations)
-      end
-      count.times do
-        FactoryGirl.create(:package, :with_set_item, :package_with_locations)
-      end
+    def create_orders
+      puts "Orders:\tCreating #{count} designated and #{count} dispatched order with orders_packages "
       count.times do
-        FactoryGirl.create(:package, :received, :with_inventory_number)
+        create_designated_packages
+        puts "\t\tCreated Order #{Order.last.id} with packages in Designated State"
+        create_dispatched_packages
+        puts "\t\tCreated Order #{Order.last.id} with packages in Dispatched State"
       end
-      count.times do
-        FactoryGirl.create(:package, :received, :with_inventory_number, :published)
+    end
+
+    def create_submitted_offer
+      FactoryGirl.create(:offer, :with_demo_items, :with_messages, created_by: donor).tap(&:submit)
+    end
+
+    def create_reviewing_offer
+      User.current_user = reviewer
+      create_submitted_offer.tap(&:start_review)
+    end
+
+    def create_reviewed_offer
+      offer = create_reviewing_offer
+      offer.update(reviewed_by: reviewer)
+      offer.tap(&:finish_review)
+    end
+
+    def create_scheduled_offer
+      create_reviewed_offer.tap(&:schedule)
+    end
+
+    def create_inactive_offer
+      create_reviewing_offer.tap(&:mark_inactive)
+    end
+
+    def create_closed_offer
+      create_reviewed_offer.tap(&:mark_unwanted)
+    end
+
+    def create_receiving_offer
+      create_reviewed_offer.tap(&:start_receiving)
+    end
+
+    def create_recieved_offer
+      inventory_offer_packages(create_receiving_offer).tap(&:receive)
+    end
+
+    def inventory_offer_packages(offer)
+      offer.reload.items.each do |item|
+        item.accept
+        item.packages.each do |package|
+          loc = Location.pluck(:id).sample
+          package.update(inventory_number: InventoryNumber.available_code, allow_web_publish: true, location_id: loc)
+          package.build_or_create_packages_location(loc, 'create')
+          package.mark_received
+        end
       end
+      offer.update(delivered_by: FactoryGirl.generate(:delivered_by))
+      offer
     end
 
     def create_single_order
-        @organisation = FactoryGirl.create(:organisation, organisation_type_id: OrganisationType.find_by_id(Random.rand(3)))
-        @processor = FactoryGirl.create(:user, :reviewer)
-        FactoryGirl.create(:order, :with_created_by, processed_by: @processor, organisation: @organisation)
+      organisation = FactoryGirl.create(:organisation, organisation_type_id: OrganisationType.find_by_id(Random.rand(3)))
```

Comment:
OrganisationType.find_by_id(Random.rand(3)) can be moved to method as it is used twice.